app/models/register.rb

```diff
@@ -79,6 +79,37 @@ def show_category_link?
     register_phase == 'Beta' && category.present?
   end
 
+  def is_register_published_by_dcms?
+    Authority.where(id: authority_id).first&.government_organisation_key === 'D5'
+  end
+
+  def fields_readable
+    register_fields.map do |field|
+      # Make the field more readable.
+      field['field_readable'] = field['field']
+                                  .humanize
+                                  .tr('-', ' ')
+
+      # GOV UK URLs are the only thing that seems to break the table on small
+      # screens. This inserts zero-width breaking spaces (`&#8203;`) to allow
+      # a URL to wrap in a narrow table cell.
+      field['text_readable'] = field['text']
+                                .gsub('.service.', '&#8203;.service.&#8203;')
+                                .html_safe
+
+      field
+    end
+  end
+
+  def safe_name
+    # `seo_title` already has 'register' in it. So we need to replace 'register
+    # register' with a 'register'.
+    [[title, seo_title, name].find(&:present?), 'register']
+      .compact
+      .join(' ')
+      .gsub('register register', 'register')
```

Comment:
We should probably check this is the end of the string, as this would do the replacement anywhere currently.

Reply:
Fixed.